rfcs/0001-restructure.md

```diff
@@ -0,0 +1,271 @@
+
+```toml
+[[order]] # HTTPD web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-httpd"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+
+[[order]] # Nginx web server
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-nginx"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+
+[[order]] # Built-in web server (web) & FPM (php-fpm)
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-dist"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/composer-install"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-fpm"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-builtin-server"
+    version = ""
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-memcached-session-handler"
+    version = ""
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/php-redis-session-handler"
+    version = ""
+    optional = true
+```
+
+## Environment Variables
+
+* `php-dist` buildpack must expose the location of the `php.ini` file through
+  an environment variable available at build and launch.
+* `php-fpm` buildpack must expose the location of the `php-fpm.conf` file through
+  an environment variable available at build and launch.
+* Session handler buildpacks must use `PHP_INI_SCAN_DIR` to set the search path
+  for additional .ini files it configures.
+
+## Unresolved Questions and Bikeshedding
+
+* php-httpd/php-nginx also sets a start command for fpm that is also set by
+  php-fpm. Does the spec allow an elegant way for start cmds set by 2
+  buildpacks to run together?
```

Comment:
For posterity only, this was answered in the working group and the answer is no. The spec prohibits doing this.